examples/customresources/demos/simplemodule/module.go

```diff
@@ -47,6 +48,13 @@ func mainWithArgs(ctx context.Context, args []string, logger logging.Logger) err
 		return err
 	}
 
+	// TODO(benji): remove this example logging.
+	time.Sleep(time.Second) // sleep to ensure module -> parent connection occurs.
+	type myStruct struct {
+		String string
+	}
+	logger.Infow("info log", "string", "stringValue", "int", 2, "float", 1.23, "bool", false, "struct", &myStruct{"string"}, "time", time.Now())
```

Comment:
This log now comes out on the RDK side like:
```
2024-02-22T10:58:56.577-0500	INFO	robot_server.simple-module	simplemodule/module.go:56	info log	{"string": "stringValue", "int": 2, "float": 1.2300000000000173, "bool": false, "struct": "&{String:string}", "time": "Local", "module_log_ts": "2024-02-22T10:58:56.574-0500"}
```

Two issues here that I wanted to get your opinion on @dgottlieb (I assume I should try to fix both, but it may require me making special cases for float and time zap field types). The first is that the float `1.23` becomes `1.2300000000000173` due to some lossy conversation either across the wire or across JSON (when `Marshaling and `Unmarshal`ing). The second is that `time.Time` values only show the timezone ("Local"), as that is the value of the `Interface` field on the `zap.Field`; the actual "time" value is encoded in the `Integer` field.